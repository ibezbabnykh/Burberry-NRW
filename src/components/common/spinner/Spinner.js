import React from 'react';

import './Spinner.scss';

const Spinner = () => (
    <div className="loadingio-spinner-rolling">
        <div className="spinner">
            <div></div>
        </div>
    </div>
);

export default Spinner;