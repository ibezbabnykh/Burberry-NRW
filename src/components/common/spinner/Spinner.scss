@import '../../../assets/style/variables';

@keyframes rotate {
    0% {
        transform: translate(-50%, -50%) rotate(0deg);
    }

    100% {
        transform: translate(-50%, -50%) rotate(360deg);
    }
}

.loadingio-spinner-rolling {
    width: 40px;
    height: 40px;
    margin: 30px auto;
    overflow: hidden;
}

.spinner {
    width: 100%;
    height: 100%;
    position: relative;
    transform: translateZ(0);
    backface-visibility: hidden;
    transform-origin: 0 0;

    div {
        box-sizing: content-box;
        position: absolute;
        width: 28px;
        height: 28px;
        border: 4px solid $color-brand;
        border-top-color: transparent;
        border-radius: 50%;
        animation: rotate 1s linear infinite;
        top: 50%;
        left: 50%;
        transform: translate(-50%, -50%);
    }
}