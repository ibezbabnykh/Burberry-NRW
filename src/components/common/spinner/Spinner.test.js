import Spinner from './Spinner';

describe('Spinner component', () => {
  it('should render', () => {
    const component = toJson(shallow(<Spinner />));
    expect(component).toMatchSnapshot();
  });
});