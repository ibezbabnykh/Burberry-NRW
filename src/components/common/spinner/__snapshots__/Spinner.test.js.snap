// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component should render 1`] = `
<div
  className="loadingio-spinner-rolling"
>
  <div
    className="spinner"
  >
    <div />
  </div>
</div>
`;
